Sources/PackageLoading/ManifestLoader.swift

```diff
@@ -355,6 +374,32 @@ public final class ManifestLoader: ManifestLoaderProtocol {
         }
     }
 
+    private func validateBinaryTargets(_ manifest: Manifest) throws {
+        // Check that binary targets point to the right file type.
+        for target in manifest.targets where target.type == .binary {
+            guard let location = URL(string: target.url ?? target.path!) else {
+                throw ManifestParseError.invalidBinaryLocation(targetName: target.name)
+            }
+
+            let isRemote = target.url != nil
+            let validSchemes = ["https"]
+            guard !isRemote || (location.scheme.map({ validSchemes.contains($0) }) ?? false) else {
+                throw ManifestParseError.invalidBinaryURLScheme(
+                    targetName: target.name,
+                    validSchemes: validSchemes
+                )
+            }
+
+            let validExtensions = isRemote ? ["zip"] : ["zip", "xcframework"]
```

Comment:
Didn't we say that we would not support zip for path-based binary deps?

Reply:
Yep, we updated this in the proposal.